cognite/client/data_classes/capabilities.py

```diff
@@ -102,34 +106,165 @@ def dump(self, camel_case: bool = False) -> dict[str, Any]:
         capability_name = self._capability_name
         return {to_camel_case(capability_name) if camel_case else to_snake_case(capability_name): data}
 
+    def has_capability(self, other: Capability) -> bool:
+        if not isinstance(self, type(other)):
+            return False
+
+        if not other.scope.is_within(self.scope):
+            return False
+        missing_actions = set(other.actions) - set(self.actions)
+        return not missing_actions
+
+
+class ProjectScope(ABC):
+    name: ClassVar[str] = "projectScope"
+
+    @classmethod
+    def load(cls, resource: dict | str) -> ProjectsScope | AllProjectsScope:
+        resource = resource if isinstance(resource, dict) else load_yaml_or_json(resource)
+        ((name, data),) = resource.items()
+        if name != cls.name:
+            raise ValueError(f"Expected '{cls.name}', got '{name}'")
+
+        if "projects" in data:
+            return ProjectsScope(projects=data["projects"])
+        elif "allProjects" in data:
+            return AllProjectsScope()
+        raise ValueError(f"Unknown project scope: {data}")
+
+    def dump(self, camel_case: bool = False) -> dict[str, Any]:
+        if isinstance(self, AllProjectsScope):
+            return {self.name: {"allProjects": {}}}
+        elif isinstance(self, ProjectsScope):
+            return {self.name: {"projects": self.projects}}
+        raise ValueError(f"Unknown project scope: {self}")
+
+
+@dataclass(frozen=True)
+class AllProjectsScope(ProjectScope):
+    ...
+
+
+@dataclass(frozen=True)
+class ProjectsScope(ProjectScope):
+    projects: list[str]
+
+
+@dataclass
+class GroupCapability:
+    capability: Capability
+
+    class Scope:
+        All = AllProjectsScope
+        Projects = ProjectScope
+
+    project_scope: ProjectScope
+
+    @classmethod
+    def load(cls, resource: dict | str) -> Self:
+        resource = resource if isinstance(resource, dict) else load_yaml_or_json(resource)
+        capability = next(key for key in resource if key != "projectScope")
+
+        return cls(
+            capability=Capability.load({capability: resource[capability]}),
+            project_scope=ProjectScope.load({ProjectScope.name: resource[ProjectScope.name]}),
+        )
+
+    def dump(self, camel_case: bool = False) -> dict[str, Any]:
+        dumped = self.capability.dump(camel_case=camel_case)
+        dumped[ProjectScope.name] = self.project_scope.dump(camel_case=camel_case)
+        return dumped
+
+
+class GroupCapabilities(UserList):
```

Comment:
Why don't we use `CogniteResourceList`?

Reply:
I was about to write a reason why not, and realized I did not agree with it, so will switch it.

Reply:
Haha 😄 Would be nice if we had a `CogniteResourceList` that is without all the nonsense (client reference, id/xid lookup etc).